backend/de.metas.banking/de.metas.banking.camt53/src/main/java/de/metas/banking/camt53/wrapper/AccountStatement2Wrapper.java

```diff
@@ -0,0 +1,246 @@
+import de.metas.currency.CurrencyCode;
+import de.metas.currency.CurrencyRepository;
+import de.metas.logging.LogManager;
+import de.metas.organization.IOrgDAO;
+import de.metas.organization.OrgId;
+import de.metas.util.Check;
+import de.metas.util.Loggables;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+import org.compiere.util.TimeUtil;
+import org.slf4j.Logger;
+
+import javax.xml.datatype.XMLGregorianCalendar;
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Optional;
+
+import static de.metas.banking.camt53.jaxb.camt053_001_02.CreditDebitCode.CRDT;
+
+@Value
+public class AccountStatement2Wrapper
+{
+	private static final Logger logger = LogManager.getLogger(AccountStatement2Wrapper.class);
+
+	@NonNull
+	IOrgDAO orgDAO;
+
+	@NonNull
+	AccountStatement2 accountStatement2;
+
+	@NonNull
+	BankAccountService bankAccountService;
+
+	@NonNull
+	CurrencyRepository currencyRepository;
+
+	@Builder
+	private AccountStatement2Wrapper(
+			@NonNull final IOrgDAO orgDAO,
+			@NonNull final AccountStatement2 accountStatement2,
+			@NonNull final BankAccountService bankAccountService,
+			@NonNull final CurrencyRepository currencyRepository)
+	{
+		this.orgDAO = orgDAO;
+		this.accountStatement2 = accountStatement2;
+		this.bankAccountService = bankAccountService;
+		this.currencyRepository = currencyRepository;
+	}
+
+	@NonNull
+	public Optional<BankStatementCreateRequest> buildBankStatementCreateRequest()
+	{
+		final BankId bankId = getBankId().orElse(null);
+
+		if (bankId == null)
+		{
+			Loggables.withLogger(logger, Level.DEBUG).addLog("Skipping AccountStatement={} because no bank was found for it!", accountStatement2);
+			return Optional.empty();
+		}
+
+		final BankAccountId bankAccountId = getBPartnerBankAccountId(bankId).orElse(null);
+		if (bankAccountId == null)
+		{
+			Loggables.withLogger(logger, Level.DEBUG).addLog(
+					"Skipping AccountStatement={} because no Bank Account was found for it!", accountStatement2);
+
+			return Optional.empty();
+		}
+
+		if (!isStatementCurrencyValid(bankAccountId))
+		{
+			Loggables.withLogger(logger, Level.WARN).addLog(
+					"Skipping AccountStatement={} because currency of the statement does not match currency of the bank account!", accountStatement2);
+			return Optional.empty();
+		}
+
+		final BigDecimal beginningBalance = getBeginningBalance();
+		final OrgId orgId = getOrgId(bankAccountId);
+		final ZoneId timeZone = orgDAO.getTimeZone(orgId);
+
+		final ZonedDateTime statementDate = getStatementDate(timeZone);
+
+		return Optional.of(BankStatementCreateRequest.builder()
+								   .orgId(orgId)
+								   .orgBankAccountId(bankAccountId)
+								   .statementDate(TimeUtil.asLocalDate(statementDate, timeZone))
+								   .name(statementDate.toString())
+								   .beginningBalance(beginningBalance)
+								   .build());
+	}
+
+	@NonNull
+	private Optional<BankId> getBankId()
```

Comment:
not sure if applies here, 
but I just want to make you aware that we have `ExplainedOptional` which basically it's an Optional but which has an `ITranslatableString explanation` in case it's empty.

Useful if you want to provide / propagate the reason why something is missing/skipped etc.

Not sure if applies in your case, but anyways, u could use it in future when needed.